util/linuxfw/nftables_runner.go

```diff
@@ -569,6 +569,12 @@ type NetfilterRunner interface {
 	// the Tailscale interface, as used in the Kubernetes egress proxies.
 	DNATNonTailscaleTraffic(exemptInterface string, dst netip.Addr) error
 
+	EnsurePortMapRuleForSvc(svc, tun string, targetIP netip.Addr, pm PortMap) error
```

Comment:
I have used 'targetIP' across all methods for services because it matches 'targetPort' else we'd have 'dst' for the tailnet target IP and 'targetPort' for tailnet target port

Reply:
That's a valid argument for the call stack to get to these methods. I think there is also an argument to be more consistent with the other methods in this interface, but I don't feel that strongly either way 👍

Reply:
> I think there is also an argument to be more consistent with the other methods in this interface

I was actually considering changing 'dst' to 'target' for that reason across the methods, but didn't do in this PR (the amount of changes is already significant). Also, we don't know yet what we want to do with this configuration mechanism in future (implement in userspace, bring into the router, etc)